airqo-platform/AirQo-api
# Airqualibrate pdt1   similar devices

# Airqualibrate pdt1 similar devices

**_WHAT DOES THIS PR DO?_**
Creates 2 APIs
1. For calibrating data from the user using the model trained with AirQo data. 
2. For training and calibrating data using a model trained with collocation data from the user. (If the user has their own data from the collocation of a low-cost and a reference monitor)

**_WHAT JIRA STORY/TASK IS RELATED TO THIS PR?_**
https://airqoteam.atlassian.net/jira/software/projects/DP/boards/44?assignee=5d91eed256bb950dd6489270&selectedIssue=DP-172

**_WHAT IS THE LINK TO THE PR BRANCH_**
https://github.com/airqo-platform/AirQo-api/tree/Airqualibrate-pdt1---similar-devices/

**_HOW DO I TEST OUT THIS PR?_**
cd AirQo-api/src/calibrate
python -m venv [DESTINATION_FOLDER]
[DESTINATION_FOLDER]/Scripts/activate
pip install -r requirements.txt

python jobs\regression.py (one time)
flask run
NB: Make sure you add a GCP service account key under jobs/      (available on request)

flask run

SAMPLE REQUEST BODY:
 See images below for sample request bodies

NB:  
1.  accepted values for the KEY "pollutant"  are PM2.5 and PM10 (this will be a radio button on the frontend)
2. for KEY "file", please upload a CSV containing data for each of the keys specified except "file" and "pollutant"
3. for the rest of the entries except where KEY is "file" or "pollutant", the KEY is the column name in the uploaded CSV corresponding to the fields given by VALUE.  i.e. datetime, sensor1 pm2.5, sensor2 pm2.5, sensor1 pm10, sensor1 pm10, temperature, humidity and collocated reference monitor PM values. 
4. SAMPLE CSV FILE AVAILABLE ON REQUEST

API 1
![api 1](https://user-images.githubusercontent.com/55631974/154980265-aeeaf348-82fa-4e18-8cca-e93967482769.PNG)

API 2

![api 2](https://user-images.githubusercontent.com/55631974/154980337-379f4b79-8537-4f4d-b786-f90f2e5d76c2.PNG)

EXPECTED RESULTS:
API 1: data in CSV format with calibrated data (PM2.5 and PM10).
API 2:data in CSV format with calibrated data (PM2.5 or PM10) depending on the choice of reference pollutant selected.

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 
[POST] http://127.0.0.1:4001/api/v1/calibrate_tool
[POST] http://127.0.0.1:4001/api/v1/train_calibrate_tool

**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**
NA
**_ARE THERE ANY RELATED PRs?_**
NA
